src/textord/edgblob.h

```diff
@@ -46,12 +46,12 @@ class OL_BUCKETS
     C_OUTLINE_LIST *start_scan() {
       for (index = 0; buckets[index].empty () && index < bxdim * bydim - 1;
         index++);
-      return &buckets[index];
+      return &buckets[index++];  //index value has not changed
     }
                                  //next non-empty bucket
     C_OUTLINE_LIST *scan_next() {
       for (; buckets[index].empty () && index < bxdim * bydim - 1; index++);
-      return &buckets[index];
+      return &buckets[index++];  //index value has not changed
```

Comment:
it is a very bad idea making post-increment here, because for next `scan_next` you may catch segfault in buckets[index] ( index == bxdim * bydim)
But in old code `scan_next(); scan_next(); scan_next();` never find the next nonempty  element (a test for this: https://godbolt.org/z/vRbwUl)
Rather than making "beautiful" one-row code - make it readable:
```
C_OUTLINE_LIST *scan_next() {
  for (;  index + 1 < bxdim * bydim ; index++)
    if(!buckets[index].empty()) return &buckets[index++]; // next position in array
  return &buckets[index]; // last position. Don't increment!
}
```
Also, avoid copy-paste:
```
C_OUTLINE_LIST *start_scan() {
  index = 0;
  return scan_next();
}
```
or if you really want one-row code
```
C_OUTLINE_LIST *start_scan() {
  index = -1; // in this case we should start from -1
  return scan_next();
}
C_OUTLINE_LIST *scan_next() {
  // first of all check index!
  for (;  index + 1 < bxdim * bydim && buckets[++index].empty() ;);
  return &buckets[index];
}
```